server/src/main/scala/org/scassandra/server/cqlmessages/request/ExecuteRequest.scala

```diff
@@ -46,7 +46,7 @@ object ExecuteRequest {
     val numberOfVariables = bodyIterator.getShort
 
     val variableValues = variableTypes.map (varType => {
-      varType.readValue(bodyIterator, VersionTwo)
+      varType.readValueWithLength(bodyIterator)(VersionTwo)
```

Comment:
Yeah, I still think there is room for improvement here.    I switched protocol version to implicit so it didn't have to be passed around when calling readValue from a Collection ColumnType.   I've now isolated all explicit protocol checks in ColumnType/CqlProtocolHelper to making calls to ProtocolVersion (added CollectionLength for that).